kalite/ab_testing/dynamic_assets.py

```diff
@@ -0,0 +1,42 @@
+from .data.groups import get_condition_by_facility_and_unit, get_grade_by_facility
+from .data.conditions import CONDITION_SETTINGS
+
+from kalite.student_testing.utils import get_current_unit_settings_value
+
+from kalite.dynamic_assets import DynamicSettingsBase, fields
+
+class DynamicSettings(DynamicSettingsBase):
+    student_grade_level = fields.IntegerField(default=0)
+    unit = fields.IntegerField(default=1)
+
+
+# modify ds in-place. Use when you're modifying ds rather than defining new dynamic settings
+def modify_dynamic_settings(ds, request=None, user=None):
+
+    user = user or request.session.get('facility_user')
+
+    if user and not user.is_teacher:
```

Comment:
We may want to add a check for the Nalanda package for now here, since it otherwise doesn't need to waste cycles on the rest.